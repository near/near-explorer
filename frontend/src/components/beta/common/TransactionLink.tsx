import * as React from "react";

import CopyToClipboard from "@explorer/frontend/components/utils/CopyToClipboard";
import Link from "@explorer/frontend/components/utils/Link";
import { shortenString } from "@explorer/frontend/libraries/formatting";
import { styled } from "@explorer/frontend/libraries/styles";

const TransactionLinkWrapper = styled("a", {
  whiteSpace: "nowrap",
});

export interface Props {
  hash: string;
}

const TransactionLink: React.FC<Props> = React.memo(({ hash }) => (
  <>
    <Link href={`/transactions/${hash}`} passHref>
      <TransactionLinkWrapper>{shortenString(hash)}</TransactionLinkWrapper>
    </Link>
    <CopyToClipboard
      text={hash}
      css={{
        marginLeft: ".3em",
        fontSize: "1.5em",
      }}
    />
  </>
));

export default TransactionLink;
