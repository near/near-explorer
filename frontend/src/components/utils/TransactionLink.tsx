import Link from "next/link";

export interface Props {
  transactionHash: string;
  children?: React.ReactNode;
}

export default ({ transactionHash, children }: Props) => (
  <Link href="/transactions/[hash]" as={`/transactions/${transactionHash}`}>
    <a className="transaction-link">
      {children || `${transactionHash.substring(0, 7)}...`}
    </a>
  </Link>
);
