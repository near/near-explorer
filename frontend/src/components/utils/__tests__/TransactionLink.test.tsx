import renderer from "react-test-renderer";

import TransactionLink from "../TransactionLink";

describe("<TransactionLink />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <TransactionLink transactionHash="H4CpspC1bqkykKG6dtCoGmLepvcyV2f2phfoqNAn5L2b" />
      )
    ).toMatchSnapshot();
  });
});
