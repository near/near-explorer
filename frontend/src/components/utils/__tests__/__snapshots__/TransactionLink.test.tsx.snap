// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionLink /> renders 1`] = `
<a
  className="transaction-link"
  href="/transactions/H4CpspC1bqkykKG6dtCoGmLepvcyV2f2phfoqNAn5L2b"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  H4CpspC...
</a>
`;
